tests/account_change_password_test.go

```diff
@@ -72,3 +118,24 @@ func changePassword(t *testing.T, passwordClient *client.CSAPI, oldPassword stri
 		StatusCode: 200,
 	})
 }
+
+func createSession(t *testing.T, unauthedClient *client.CSAPI, userID, password string) *client.CSAPI {
+	authedClient := unauthedClient
```

Comment:
Make a new client instead of accepting an unauthed client. You're passing this client around by pointer so it could become very confusing and is likely not what you intended:
```go
unauthedClient := deployment.Client(t, "hs1", "")
sessionTest := createSession(t, unauthedClient, "test_change_password_user", "superuser")
// sessionTest == unauthedClient
```
so any requests using unauthedClient *will have auth*. To fix this, replace `unauthedClient` with `deployment` from the function signature and instead do `unauthedClient := deployment.Client(t, "hs1", "")`

Reply:
Got it 👍